### ISSUE TITLE
Zero-unit rule should use units for flex shorthand property [for IE10/IE11]

### ISSUE THREAD

**Issue by CASandmann** [sasstools/sass-lint#1223]
**What version of Sass Lint are you using?**
1.12.1

**What did you do? Please include the actual source code causing the issue.**
I used sass-lint-auto-fix on the zero-unit rule, but that's just following the rules.  It removed all units from the `flex` shorthand properties that were 0 on the last argument (the flex-basis part).
```
flex: 1 0 0px;
```
becomes
```
flex: 1 0 0;
```

**What did you expect to happen?**
This causes [a known issue with IE10 & IE11](https://philipwalton.com/articles/normalizing-cross-browser-flexbox-bugs/) where it treats the entire property as invalid.  The obvious "proper" way to conform to the lint rule is to remove the unit which causes incompatibilities with older browsers which may need to be supported.  A current workaround for this rule is to use `0%` as the last property which is valid for IE and doesn't trigger the lint rule.

I think an exception to the lint rule needs to be added for the `flex` shorthand property, or at least add a configuration option to force a unit so developers can ensure compatibility with older browsers if they need to support it.

**If you're using a IDE plugin have you tried the CLI too?**
Both the IDE plugin and the CLI give the error.
